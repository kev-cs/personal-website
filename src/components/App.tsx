import * as React from "react";
import { IntlProvider } from "react-intl";
import Header from "components/Header";
import Home from "components/Home";
import LanguageSelector from "components/LanguageSelector";

import fr from "assets/translations/fr.json";

type State = {
  locale: string;
}

const supportedLanguagesBackingArray = ["en", "fr"] as const;
const supportedLanguages: Array<string> = [...supportedLanguagesBackingArray];
type SupportedLanguage = typeof supportedLanguagesBackingArray[number];

export default class App extends React.Component <{}, State> {

  private readonly defaultLocale: SupportedLanguage = "en";
  private readonly translations: Record<SupportedLanguage, Record<string, string>> = {
    en: null, // default locale used in application
    fr
  };

  state = {
    locale: navigator.language.split(/[-_]/)[0]  // locale without region code
  };

  changeLanguage = (lang: string): void => {
    this.setState({
      locale: lang
    });
  };

  render(): JSX.Element {
    const locale: SupportedLanguage = supportedLanguages.includes(this.state.locale) ?
      this.state.locale as SupportedLanguage : this.defaultLocale;

    const Footer = (): JSX.Element => (
      <footer id="footer"><span>Â© 2019 Kevin Caro Silva</span></footer>
    );

    return (
      <>
        <IntlProvider locale={locale} messages={this.translations[locale]} defaultLocale={this.defaultLocale}>
          <LanguageSelector changeLanguage={this.changeLanguage} supportedLanguages={supportedLanguages}/>
          <Header/>
          <Home/>
          <Footer/>
        </IntlProvider>
      </>
    );
  }
}

