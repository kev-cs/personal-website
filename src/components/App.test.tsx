import App from "./App";
import { create } from "react-test-renderer";
import React from "react";
import "@testing-library/jest-dom/extend-expect";

describe("App", () => {

  it("renders correctly", () => {
    const element = create(<App/>);

    expect(element.toJSON()).toMatchSnapshot();
  });

});
