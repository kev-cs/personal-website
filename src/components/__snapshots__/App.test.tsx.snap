// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Array [
  <div
    id="webDevNotice"
  >
    <i
      className="fas fa-exclamation-circle"
    />
    <div>
      <h2>
        Slowly but surely
      </h2>
      <p>
        I'm currently developing this website, more coming soon!
      </p>
    </div>
  </div>,
  <div
    id="languageSelector"
  >
    <i
      className="fas fa-language inactive"
      data-testid="toggle-show-languages"
      onClick={[Function]}
    />
  </div>,
  <header
    id="header"
  >
    <div>
      <img
        alt="Personal website logo"
        src="test-file-stub"
      />
      <h1
        id="name"
      >
        Kevin Caro Silva
      </h1>
      <h2
        id="profession"
      >
        Software developer
      </h2>
    </div>
    <div
      id="externalLinks"
    >
      <a
        href="https://github.com/kev-cs"
      >
        <i
          className="fab fa-github"
        />
      </a>
      <a
        href="https://www.linkedin.com/in/kevin-caro-silva-36670a136/"
      >
        <i
          className="fab fa-linkedin"
        />
      </a>
      <a
        href="mailto:admin@kevincs.me"
      >
        <i
          className="fas fa-envelope"
        />
      </a>
    </div>
  </header>,
  <div
    id="home"
  >
    <header
      id="greeting"
    >
      Hi, I'm a software developer based in Montreal, QC.
      <hr
        id="greetingHr"
      />
    </header>
    <p>
      In a few words : Java, Agile, web development, automation, and simplicity.
       
      I'm also a fan of clean coding, Linux, cloud development, CI/CD, TDD, and too much more to enumerate. I enjoy exploring solutions for software design and architecture.
       
    </p>
    <p>
      Currently doing Java development at Desjardins Group.
    </p>
  </div>,
  <footer
    id="footer"
  >
    <span>
      Â© 2019 Kevin Caro Silva
    </span>
  </footer>,
]
`;
